controllers/profile.js

```diff
@@ -0,0 +1,28 @@
+const mongoose = require('mongoose');
+const { logger } = require('../logging/logger');
+
+const Users = mongoose.model('User');
+
+// WARNING any schema properties in this object will be exposed to other clients,
+// do not put sensitive schema properties in this object
+const PUBLIC_CONTACT_SCHEMA = {
+    person: 1,
+    demo: 1,
+    gen_interest: 1,
+    spec_interests: 1,
+    user_type: 1,
+};
+
+exports.getProfileData = function (req, res) {

```

Comment:
For the get profile and get contacts, we want to maintain the restriction that a user can only get the profile info for themselves or someone in their contacts, and can only get the contact info for themselves. We don't consider the userId to be a secret, and we don't want somebody to be able to see the SAT scores, GPA, etc. of another student they aren't "connected" with.

It looks like in the authorize call we look up by the userId and api-key. I just want to double check that that lookup will provide for the above restrictions.

Reply:
getProfile is fine - the only thing we need to do is check if the auth key matches the requested user id. getContacts is trickier, which is why I introduced the public contact schema. The populate command in mongoose is the key player here, it enables you to only populate the db response with a certain subset of data. I have a test for this ensuring we only pass the correct public info described in the `PUBLIC_CONTACT_SCHEMA` in this file

Reply:
Oh I see. Is there a way for a user to get the non-public contact data for display? For a mentor, they need to be able to see all the info that their mentee would get from calling getProfile. A mentor is the only user that should be able to see all of that data. 

Our app UI mock-up is here, which might make a little more sense. 
https://www.figma.com/file/2TOYmQtfx3HTq11em2wZ81/Hey-Mentor---vNext?node-id=0%3A1

Reply:
Ah okay, gimme a bit to figure out the most RESTful way to do this

Reply:
There's no current plans to have those types of relationships, although ideally this API would be agnostic to the type of user, and would be restricted more by who is in the contacts and who isn't.